crates/external_services/src/grpc_client.rs

```diff
@@ -1,18 +1,38 @@
 /// Dyanimc Routing Client interface implementation
 #[cfg(feature = "dynamic_routing")]
 pub mod dynamic_routing;
+/// gRPC based Heath Check Client interface implementation
+#[cfg(feature = "dynamic_routing")]
+pub mod health_check_client;
 use std::{fmt::Debug, sync::Arc};
 
 #[cfg(feature = "dynamic_routing")]
 use dynamic_routing::{DynamicRoutingClientConfig, RoutingStrategy};
+#[cfg(feature = "dynamic_routing")]
+use health_check_client::HealthCheckClient;
+#[cfg(feature = "dynamic_routing")]
+use http_body_util::combinators::UnsyncBoxBody;
+#[cfg(feature = "dynamic_routing")]
+use hyper::body::Bytes;
+#[cfg(feature = "dynamic_routing")]
+use hyper_util::client::legacy::connect::HttpConnector;
 use serde;
+#[cfg(feature = "dynamic_routing")]
+use tonic::Status;
+
+#[cfg(feature = "dynamic_routing")]
+/// Hyper based Client type for maintaining connection pool for all gRPC services
+pub type Client = hyper_util::client::legacy::Client<HttpConnector, UnsyncBoxBody<Bytes, Status>>;
 
 /// Struct contains all the gRPC Clients
 #[derive(Debug, Clone)]
 pub struct GrpcClients {
     /// The routing client
     #[cfg(feature = "dynamic_routing")]
     pub dynamic_routing: RoutingStrategy,
+    /// Health Check client for all gRPC services
+    #[cfg(feature = "dynamic_routing")]
+    pub health_client: HealthCheckClient,
 }
```

Comment:
According to this type, you will still have to build 2 connections (one for actual service communication and other for health check)? Should this health check client be tied to the actual gRPC services types?
cc: @Aprabhat19